x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/get_number_of_items.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HeadlessChromiumDriver as HeadlessBrowser } from '../../../../server/browsers/chromium/driver';
+import { LevelLogger as Logger } from '../../../../server/lib/level_logger';
+import { LayoutInstance } from '../../layouts/layout';
+
+export const getNumberOfItems = async (
+  browser: HeadlessBrowser,
+  layout: LayoutInstance,
+  logger: Logger
+): Promise<number> => {
+  logger.debug('determining how many items we have');
```

Comment:
This is new for this function: previously the log call was happening in a different part of the Rx chain

Reply:
If not, we will be soon. I'm working in a branch where logging (hopefully) makes more sense and gives `info` level details about what's going on :)

Reply:
We'll log it from `waitForElementsToBeInDOM`

- [debug] first log how many elements it's waiting for,
- [info] finish by logging how many elements it found